src/query_plan/operation.rs

```diff
@@ -869,6 +891,34 @@ pub(crate) mod normalized_field_selection {
             &mut self.data.sibling_typename
         }
 
+        pub(crate) fn type_if_added_to(
+            &self,
+            parent_type: &CompositeTypeDefinitionPosition,
+            schema: &ValidFederationSchema,
+        ) -> Option<CompositeTypeDefinitionPosition> {
+            let data = self.data();
+            if &data.field_position == parent_type {
+                let digest = match &self.data.field_position {
+                    FieldDefinitionPosition::Object(ty) => CompositeTypeDefinitionPosition::Object(
+                        ObjectTypeDefinitionPosition::new(ty.type_name.clone()),
+                    ),
+                    FieldDefinitionPosition::Interface(ty) => {
+                        CompositeTypeDefinitionPosition::Interface(
+                            InterfaceTypeDefinitionPosition::new(ty.type_name.clone()),
+                        )
+                    }
+                    FieldDefinitionPosition::Union(ty) => CompositeTypeDefinitionPosition::Union(
+                        UnionTypeDefinitionPosition::new(ty.type_name.clone()),
+                    ),
+                };
+                return Some(digest);
+            }
+            if data.name() == parent_type.type_name() {
+                return Some(parent_type.introspection_typename_field().parent());
+            }
+            self.can_rebase_on(parent_type, schema)
+                .then(|| data.field_position.parent())
```

Comment:
this should be returning actual field type -> note that field parent is different than the parent (first check should be checking whether field is on the same parent equivalence), e.g. it can be from different schema so this should be doing something along those lines

```rs
// get parent field -> get field from schema -> get type
parent_type.field(data.field_position.field_name())?.get(schema.schema())?.ty
```